### PR TITLE
WIP: Swift 3 support

### PR DESCRIPTION
**Pull request by serejahh**
- [x] Migrate to Swift 3 syntax
- [x] Update API
- [ ] Make tests pass
- [ ] Review the PR by @netguru team
- [ ] 🥂

### CODE REVIEW

**Comment by akashivskyy** [comment]
Hi @serejahh! How is the progress going? Maybe you could use some help? 😉

**Comment by serejahh** [comment]
Hi @akashivskyy. I hope I'll finish 2 step on the weekend. If you have time, let's finish it together 😄

**Comment by akashivskyy** [comment]
I'm happy to help! 😊

I created a `feature/swift-3-dev` branch so that we can collaborate without touching `develop`. You can re-target (close and one another) this pull request to `feature/swift-3-dev` and we'll work together on stage 2. 😉

**Comment by akashivskyy** [comment]
Nevermind that reopening, seems like I can change the base just like that. 😄

**Comment by serejahh** [comment]
That's great! I've reviewed API and I can't say it doesn't follow. I just changed one thing

**Comment by serejahh** [comment]
About tests:
1. I've fixed tests for JSONDeserializer, but the solution doesn't look good 😿. Please have a look at https://github.com/netguru/ResponseDetective/pull/18/commits/a3e67c5b29109cc1ba3b80e4396a7a87bc044760
2. What would your suggestion be for tests which compare buffer output just by comparing strings (I mentioned it in #13)?

**Comment by akashivskyy** [comment]
As for conforming to [Swift 3's API Design Guidelines](https://swift.org/documentation/api-design-guidelines/), I see lots of places that can be improved in addition to changes in 707d1ff.

As for fixing `JSONBodyDeserializer` tests, I think your solution would be temporarily okay, just remember to **use tabs, not spaces** as we follow [GitHub's Style Guide](https://github.com/github/swift-style-guide) (I will put a link to that in README and CONTRIBUTING files).

As for fixing the buffer comparison tests, I think this is out of scope of this superb pull request. I'll need to think that over. 😉

Let me know if this is ready for in-depth code review. ✨

**Comment by serejahh** [comment]
Oh, sorry. I've just fixed it 😉 

I believe you can review and merge it (to be able to commit your change)

File: ResponseDetective.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -873,55 +873,63 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 3A7D01861CD3BE0100B7FBD7 /* Framework-macOS-Debug.xcconfig */;
 			buildSettings = {
+				SWIFT_VERSION = 3.0;
```
Comment (akashivskyy) [inline ResponseDetective.xcodeproj/project.pbxproj]: Please remove these `SWIFT_VERSION` settings from the project file itself (by hitting backspace in "Build Settings" list) and update the value in [ResponseDetective-Common.xcconfig](https://github.com/netguru/ResponseDetective/blob/develop/Configuration/ResponseDetective-Common.xcconfig) instead.

File: Sources/BufferOutputFacility.swift [hunk 2]
```diff
@@ -11,32 +11,32 @@ import Foundation
 @objc(RDTBufferOutputFacility) public final class BufferOutputFacility: NSObject, OutputFacility {
 
 	/// A buffer of request representations.
-	public private(set) var requestRepresentations: [RequestRepresentation] = []
+	public fileprivate(set) var requestRepresentations: [RequestRepresentation] = []
```
Comment (akashivskyy) [inline Sources/BufferOutputFacility.swift]: Let's use `private` if possible _(this affects other `fileprivate`s as well)_.

File: ResponseDetective.playground/Contents.swift [hunk 3]
```diff
@@ -19,18 +19,18 @@ XCPlaygroundPage.currentPage.needsIndefiniteExecution = true
...
+let session = URLSession(configuration: configuration)
 
 //: ---
 
 // Now, just create a request and resume its session data task.
 
-let request = NSURLRequest(URL: NSURL(string: "https://httpbin.org/get")!)
-session.dataTaskWithRequest(request).resume()
+let request = URLRequest(url: URL(string: "https://httpbin.org/get")!)
+session.dataTask(with: request).resume()
```
Comment (akashivskyy) [inline ResponseDetective.playground/Contents.swift:34]: Thanks for remembering about the playground! ✨

File: Tests/JSONBodyDeserializerSpec.swift [hunk 4]
```diff
@@ -19,10 +19,12 @@ internal final class JSONBodyDeserializerSpec: QuickSpec {
...
 
 			it("should correctly deserialize JSON data") {
-				let source = ["foo": "", "bar": 0, "baz": false, "qux": [AnyObject](), "corge": [String: AnyObject]()]
-				let data = try! NSJSONSerialization.dataWithJSONObject(source, options: [])
-				let expected = try! NSString(data: NSJSONSerialization.dataWithJSONObject(source, options: [.PrettyPrinted]), encoding: NSUTF8StringEncoding)!
-				expect { sut.deserializeBody(data) }.to(equal(expected))
+                let source: [String: Any] = ["foo": "", "bar": 0, "baz": false, "qux": [AnyObject](), "corge": [String: Any]()]
+				let data = try! JSONSerialization.data(withJSONObject: source, options: [])
+                let actualData = sut.deserializeBody(data)!.data(using: .utf8)!
+                let actualJSON = try! JSONSerialization.jsonObject(with: actualData, options: []) as! [String: Any]
```
Comment (akashivskyy) [inline Tests/JSONBodyDeserializerSpec.swift]: You forgot to change spaces to tabs here... 🙈

File: Tests/ResponseDetectiveSpec.swift [hunk 5]
```diff
@@ -64,10 +64,12 @@ internal final class ResponseDetectiveSpec: QuickSpec {
 
 					beforeEach {
 						ResponseDetective.ignoreRequestsMatchingPredicate(NSPredicate { subject, _ in
-							guard let subject = subject as? NSURLRequest, let URL = subject.URL, let string = URL.absoluteString else {
+							guard let subject = subject as? URLRequest, let url = subject.url else {
 								return true
 							}
-							return string.containsString("foo")
+                            let string = url.absoluteString
```
Comment (akashivskyy) [inline Tests/ResponseDetectiveSpec.swift]: And here as well (tabs). 😄

**Review by akashivskyy — CHANGES_REQUESTED** [review]
This looks very very good! 👍

In addition to the comments below, could you also update `README.md` to reflect the changes you've made? 😊

File: Tests/JSONBodyDeserializerSpec.swift (hunk 4 above)
Comment (serejahh) [reply Tests/JSONBodyDeserializerSpec.swift]: Damn! Github diff didn't show

File: Sources/BufferOutputFacility.swift (hunk 2 above)
Comment (serejahh) [reply Sources/BufferOutputFacility.swift]: Why do you prefer `private` over `fileprivate`? I love to use extensions in ways [Natasha The Robot](https://www.natashatherobot.com/using-swift-extensions/) described. In this case you need `fileprivate`

File: ResponseDetective.xcodeproj/project.pbxproj (hunk 1 above)
Comment (serejahh) [reply ResponseDetective.xcodeproj/project.pbxproj]: image
Looks dangerous 😄

File: Sources/BufferOutputFacility.swift (hunk 2 above)
Comment (akashivskyy) [reply Sources/BufferOutputFacility.swift]: Yes, but I don't see any extensions. 😉 

I just prefer the most restrictive access modifier – if there are no extensions, there is nothing that justifies them to be `fileprivate`.

File: ResponseDetective.xcodeproj/project.pbxproj (hunk 1 above)
Comment (akashivskyy) [reply ResponseDetective.xcodeproj/project.pbxproj]: As it should. 😄

**Review by akashivskyy — APPROVED** [review]
Great work! 🚢

**Pull request merged**